app/controllers/WhyCannotPayInFullController.scala

```diff
@@ -81,20 +85,44 @@ class WhyCannotPayInFullController @Inject() (
       formWithErrors => Ok(views.whyCannotPayInFull(formWithErrors)),
       { reasons =>
 
-        journeyConnector.updateWhyCannotPayInFullAnswers(
-          request.journeyId,
-          WhyCannotPayInFullAnswers.WhyCannotPayInFull(reasons)
-        ).map(updatedJourney =>
+        updateJourney(request.journey, existingAnswersInJourney(request.journey), reasons)
+          .map{ updatedJourney =>
             Routing.redirectToNext(
               routes.WhyCannotPayInFullController.whyCannotPayInFull,
               updatedJourney,
               //even if the value is changed we still want to go to cya page
               submittedValueUnchanged = true
-            ))
+            )
+          }
       }
     )
   }
 
+  private def updateJourney(
+      journey:    Journey,
+      oldAnswers: Option[Set[CannotPayReason]],
+      newAnswers: Set[CannotPayReason]
+  )(implicit r: AuthenticatedJourneyRequest[_]): Future[Journey] =
+    if (oldAnswers.contains(newAnswers)) {
```

Comment:
just the smallest thing; what if the new answers have less than the old? say originally they have 2 answers and then deselect one in the change, old answers would still contain new answers, but we wouldn't register that change

Reply:
the `.contains` is on the `Option` rather than on the underlying `Set`, so we should only get a `true` if the `Option` is defined and the underlying `Set`'s is equal to `newAnswers`

Reply:
ah that makes sense. cheers